xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -1792,4 +1750,13 @@ String print(MessageOrBuilder message) {
       return res;
     }
   }
+
+  @VisibleForTesting
+  static final class InvalidProtoDataException extends RuntimeException {
+    private static final long serialVersionUID = 1L;
+
+    private InvalidProtoDataException(String message) {
+      super(message);
```

Comment:
Might better to disable stacktrace which causes a lot of memory allocation.

Reply:
Yes, thanks for the suggestion.